Dockerfile

```diff
@@ -0,0 +1,35 @@
+# Use the official ROS 2 humble base image
+FROM ros:humble
+
+# Set the working directory
+WORKDIR /ros2_workspace/src
+
+COPY . .
+# Install additional dependencies
+RUN apt-get update \
+    && apt-get -y --quiet --no-install-recommends install \
+    gcc \
+    git \
+    libxml2-dev \
+    libxslt-dev \
+    python3 \
+    ros-humble-ament-cmake-clang-format\
+    ros-humble-rviz2
+
+# Set environment variables
+ENV ROS_DISTRO humble
+ENV ROS_VERSION 2
+WORKDIR /ros2_workspace
+RUN . /opt/ros/humble/setup.sh
+RUN rosdep update
+RUN rosdep install --from-paths src --ignore-src -y
+
+RUN . /opt/ros/humble/setup.sh && colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release --packages-up-to grid_map_geo
```

Comment:
Would you like to split this into a multi-stage dockerfile like we do here? 
https://github.com/ethz-asl/terrain-navigation/pull/34/files

This allows developers to have a layer with all the dependencies needed, volume mount the source code, run colcon build every time they make changes.

Reply:
Can you do it for me :)) I dant understand what I do

Reply:
Yep!

Reply:
https://github.com/avcuenes/grid_map_geo/pull/1